core/src/main/java/com/linecorp/armeria/client/AttributeMapMaybeWithParent.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client;
+
+import java.util.Iterator;
+
+import io.netty.util.Attribute;
+import io.netty.util.AttributeKey;
+import io.netty.util.AttributeMap;
+
+/**
+ * Holds {@link Attribute}s which can be accessed via {@link AttributeKey}.
+ */
+// TODO(minwoox) Better name for this?
+public interface AttributeMapMaybeWithParent extends AttributeMap {
+
+    /**
+     * Returns the {@link Iterator} of all {@link Attribute}s this map contains. If this
+     * {@link AttributeMapMaybeWithParent} has its parent, then the {@link Iterator} will iterate it afterward.
+     *
+     * @see #attrs(boolean)
+     */
+    Iterator<Attribute<?>> attrs();
+
+    /**
+     * Returns the {@link Iterator} of all {@link Attribute}s this map contains.
+     *
+     * @param withAttrsInParent tells whether to iterate the parent's {@link Attribute}s after iterating this
+     *                          {@link AttributeMapMaybeWithParent}'s {@link Attribute}s
+     */
+    Iterator<Attribute<?>> attrs(boolean withAttrsInParent);
```

Comment:
How about `ownAttrs()`? instead of using a flag? (Named after JavaScript `hasOwnPropertyNames()`)